sdk/src/Core/Amazon.Runtime/Internal/Auth/AWS4Signer.cs

```diff
@@ -893,16 +898,7 @@ protected internal static string CanonicalizeHeaders(IEnumerable<KeyValuePair<st
         /// <returns>Formatted string of header names</returns>
         protected static string CanonicalizeHeaderNames(IEnumerable<KeyValuePair<string, string>> sortedHeaders)
         {
-            var builder = new StringBuilder();
-            
-            foreach (var header in sortedHeaders)
-            {
-                if (builder.Length > 0)
-                    builder.Append(";");
-                builder.Append(header.Key.ToLowerInvariant());
-            }
-            
-            return builder.ToString();
+            return string.Join(";", sortedHeaders.Select(x => x.Key.ToLowerInvariant()));
```

Comment:
Ran another benchmark on Fedora and decided to switch to ValueStringBuilder here too

![image](https://github.com/aws/aws-sdk-net/assets/174258/9b578759-1376-44de-856a-110c4b1a3cdf)